src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEFieldSymbol.cs

```diff
@@ -276,7 +298,11 @@ private void EnsureSignatureIsLoaded()
                 type = NullableTypeDecoder.TransformType(type, _handle, moduleSymbol, accessSymbol: this, nullableContext: _containingType);
                 type = TupleTypeDecoder.DecodeTupleTypesIfApplicable(type, _handle, moduleSymbol);
 
-                _lazyIsVolatile = customModifiersArray.Any(m => !m.IsOptional && ((CSharpCustomModifier)m).ModifierSymbol.SpecialType == SpecialType.System_Runtime_CompilerServices_IsVolatile);
+                RefKind refKind = fieldInfo.IsByRef ?
+                    moduleSymbol.Module.HasIsReadOnlyAttribute(_handle) ? RefKind.RefReadOnly : RefKind.Ref :
```

Comment:
>HasIsReadOnlyAttribute [](http://example.com/codeflow?start=40&length=22)

It doesn't look like the specification explicitly mentions this attribute.

Reply:
Clarified the behavior here https://github.com/dotnet/csharplang/pull/6042/commits/1c2b455db989372cecf0564474e9a11c7ed5db4a